.circleci/real_config.yml

```diff
@@ -5098,6 +5123,181 @@ workflows:
+
+      # Singularity over SLURM test on GCP
+      - test-e2e-hpc-gcp:
+          context:
+            # Provides the GITHUB_USERNAME and GITHUB_TOKEN enviroment variable
+            # that's required by the "gh" command for authentication.
+            - github-read
+            - gcp
+          matrix:
+            parameters:
+              name: [test-e2e-slurm-singularity-gcp]
+              always-run: [true]
+              mark: ["e2e_slurm and not parallel and not gpu_required"]
+          requires:
+            - build-go
+            - request-hpc-tests
+
+      # Podman over SLURM test on GCP
+      - test-e2e-hpc-gcp:
+          context:
+            # Provides the GITHUB_USERNAME and GITHUB_TOKEN enviroment variable
+            # that's required by the "gh" command for authentication.
+            - github-read
+            - gcp
+          matrix:
+            parameters:
+              name: [test-e2e-slurm-podman-gcp]
+              container-run-type: ["podman"]
+              mark: ["e2e_slurm and not parallel and not gpu_required"]
+          requires:
+            - build-go
+            - request-hpc-tests
+
+      # Enroot over SLURM test on GCP
+      - test-e2e-hpc-gcp:
+          context:
+            # Provides the GITHUB_USERNAME and GITHUB_TOKEN enviroment variable
+            # that's required by the "gh" command for authentication.
+            - github-read
+            - gcp
+          matrix:
+            parameters:
+              name: [test-e2e-slurm-enroot-gcp]
+              container-run-type: ["enroot"]
+              mark: ["e2e_slurm and not parallel and not gpu_required"]
+          requires:
+            - build-go
+            - request-hpc-tests
+
+      # Singularity over PBS test on GCP
+      - test-e2e-hpc-gcp:
+          context:
+            # Provides the GITHUB_USERNAME and GITHUB_TOKEN enviroment variable
+            # that's required by the "gh" command for authentication.
+            - github-read
+            - gcp
+          matrix:
+            parameters:
+              name: [test-e2e-pbs-singularity-gcp]
+              workload-manager: ["pbs"]
+              always-run: [true]
+              mark: ["e2e_pbs and not parallel and not gpu_required"]
+          requires:
+            - build-go
+            - request-hpc-tests
+
+      # Podman over PBS test on GCP
+      - test-e2e-hpc-gcp:
+          context:
+            # Provides the GITHUB_USERNAME and GITHUB_TOKEN enviroment variable
+            # that's required by the "gh" command for authentication.
+            - github-read
+            - gcp
+          matrix:
+            parameters:
+              name: [test-e2e-pbs-podman-gcp]
+              container-run-type: ["podman"]
+              workload-manager: ["pbs"]
+              mark: ["e2e_pbs and not parallel and not gpu_required"]
+              extra-pytest-flags: ["-k 'not test_slurm_verify_home'"]
+          requires:
+            - build-go
```

Comment:
![image](https://github.com/determined-ai/determined/assets/31451191/9923af2d-daa0-4ed3-bb46-7318b0b70b58)

I think the dependencies are mixed up? build-go is behind ```test-perf-feature``` branch so that it doesn't run on every commit twice

package and publish system is behind the package test requests.

Ideally I think approving the hpc tests should be enough to run them.

Reply:
I think the dependencies can be fixed pretty easily, but I'm not sure what you meant by build-go would get run twice on a commit. Would it be an issue just to move it outside of the request?

Reply:
For example ```build-react``` runs in ```e2e-tests``` and also in the ```test-e2e-longrunning```. In a perfect world we would only build go / react once per commit. These tests run on every commit a lot so it is wasteful. 

I don't really know an easy solution besides duplicating build go so if you just want to move it outside of the request that is fine too.

Reply:
@NicholasBlaskey  sorry got caught up in release stuff. I think I fixed the dependencies issue and updated